April_report.md

```diff
@@ -0,0 +1,366 @@
+          autocomplete="off"
+          placeholder="What needs to be done?"
+          @keyup.enter="addTodo">
+      </header>
+      <!-- main section -->
+      <section class="main" v-show="todos.length">
+        <input class="toggle-all" id="toggle-all"
+          type="checkbox"
+          :checked="allChecked"
+          @change="toggleAll(!allChecked)">
+        <label for="toggle-all"></label>
+        <ul class="todo-list">
+          <TodoItem
+            v-for="(todo, index) in filteredTodos"
+            :key="index"
+            :todo="todo"
+          />
+        </ul>
+      </section>
+      <!-- footer -->
+      <footer class="footer" v-show="todos.length">
+        <span class="todo-count">
+          <strong>{{ remaining }}</strong>
+          {{ remaining | pluralize('item') }} left
+        </span>
+        <ul class="filters">
+          <li v-for="(val, key) in filters">
+            <a :href="'#/' + key"
+              :class="{ selected: visibility === key }"
+              @click="visibility = key">{{ key | capitalize }}</a>
+          </li>
+        </ul>
+        <button class="clear-completed"
+          v-show="todos.length > remaining"
+          @click="clearCompleted">
+          Clear completed
+        </button>
+      </footer>
+    </section>
+  </div>
+</template>
+
+<script>
+import { mapActions } from 'vuex'
+import TodoItem from './components/TodoItem.vue'
+const filters = {
+  all: todos => todos,
+  active: todos => todos.filter(todo => !todo.done),
+  completed: todos => todos.filter(todo => todo.done)
+}
+export default {
+  components: { TodoItem },
+  data () {
+    return {
+      visibility: 'all',
+      filters: filters
+    }
+  },
+  computed: {
+    todos () {
+      return this.$store.state.todos
+    },
+    allChecked () {
+      return this.todos.every(todo => todo.done)
+    },
+    filteredTodos () {
+      return filters[this.visibility](this.todos)
+    },
+    remaining () {
+      return this.todos.filter(todo => !todo.done).length
+    }
+  },
+  methods: {
+    ...mapActions([
+      'toggleAll',
+      'clearCompleted'
+    ]),
+    addTodo (e) {
+      const text = e.target.value
+      if (text.trim()) {
+        this.$store.dispatch('addTodo', text)
+      }
+      e.target.value = ''
+    }
+  },
+  filters: {
+    pluralize: (n, w) => n === 1 ? w : (w + 's'),
+    capitalize: s => s.charAt(0).toUpperCase() + s.slice(1)
+  }
+}
+</script>
+
+<style src="todomvc-app-css/index.css"></style>
+```
+
+* Thêm css cho project
+
+Để làm cho project đẹp hơn mình đã sử dụng todomvc-app-css, link hướng dẫn mình để dưới nha =))
+
+[todomvc-app-css](https://github.com/tastejs/todomvc-app-css)
+
+### Done! Mở terminal ngay và luôn và xem thành quả
+
+```html
+$ npm run dev
+```
+
+![aa](https://images.viblo.asia/67c0bf6c-1363-48f4-992a-84ddb63ffaaa.png)
+
+Qua bài viết chắc rằng chúng ta cũng có thêm những hiểu viết về Vuex và áp dụng hiệu quả cho các dự án của mình.
```

Comment:
E viết toàn câu nghe rất trống không nhé. Nó không tóm được ý muốn truyền tải ý. Đọc cứ thấy kiểu cộc lốc, hoặc lửng ý.